### ISSUE TITLE
Inconsistent Numeric Types (Decimal vs. Float)

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#278]
## Description

Pervasive mixing of `decimal.Decimal` and `float` types across the codebase results in lost precision for GAAP-compliant financial calculations.

The core financial models (`manufacturer.py` and `ledger.py`) correctly use `Decimal` for precise financial calculations, but the Monte Carlo simulation layer (`monte_carlo_worker.py`, `simulation.py`) and analysis modules aggressively cast values to `float` for numpy compatibility. This undermines the precision benefits of using `Decimal` in the first place.

## Problem Statement

The precision gained by `Decimal` in the core model is effectively lost when the simulation loop converts `revenue` to `float` for ratio calculations (e.g., `revenue_multiplier`). This adds computational overhead without delivering guaranteed precision in the final results.

**Key Issue**: `Decimal` is a hard requirement for auditing and GAAP compliance, so the simulation loop and worker must maintain `Decimal` typing until the final aggregation step.

## Context

- **Project Area**: Core simulation architecture
- **Complexity Score**: 6-7 (architectural change affecting multiple core modules)
- **Priority**: High - affects financial accuracy guarantees

### Current Type Flow (Problematic)

```
[manufacturer.py]           [monte_carlo_worker.py]          [Results]
Decimal fields      →       float() conversion       →       numpy float64
- total_assets              - line 83: float(revenue)        - final_assets
- revenue                   - line 85-86: float(premium)     - annual_losses
- equity                    - line 106: float(sum(losses))   - insurance_recoveries
```

### Files Correctly Using Decimal (Reference)

1. **`ergodic_insurance/ledger.py`** - Full Decimal implementation for double-entry accounting
2. **`ergodic_insurance/manufacturer.py`** - Uses `to_decimal()` for all financial fields
3. **`ergodic_insurance/decimal_utils.py`** - Utility functions (`to_decimal`, `ZERO`, `quantize_currency`)
4. **`ergodic_insurance/financial_statements.py`** - Financial statement generation
5. **`ergodic_insurance/accrual_manager.py`** - Accrual tracking with Decimal

### Files Requiring Refactoring

1. **`ergodic_insurance/monte_carlo_worker.py`** (Primary)
   - Line 83: `revenue_multiplier = float(revenue) / initial_revenue`
   - Line 85-86: `annual_premium = float(insurance_program.calculate_annual_premium()) * revenue_multiplier`
   - Line 98: `revenue=float(revenue)` passed to loss generator
   - Line 106: `total_year_loss = float(sum(loss.amount for loss in year_losses))`
   - Line 113-115: `float()` conversions for recovery and retained amounts
   - Line 142: `float(sim_manufacturer.total_assets)` for final results

2. **`ergodic_insurance/simulation.py`** (Secondary)
   - Uses numpy arrays (`np.zeros`) which default to float64
   - Pre-allocates float arrays for results: `self.assets`, `self.equity`, `self.roe`, etc.

3. **`ergodic_insurance/business_optimizer.py`** (Analysis Layer)
   - ~50 occurrences of `float()` conversions
   - Lines 223-224, 349, 549, 627, 737-738, 762-763, etc.

4. **`ergodic_insurance/decision_engine.py`** (Analysis Layer)
   - Lines 701, 752, 756, 793, 801, 820, 822, etc.

5. **`ergodic_insurance/ergodic_analyzer.py`** (Analysis Layer)
   - Lines 780, 1512-1514, 2267, 2336-2337

### R1: Maintain Decimal Precision Through Simulation Loop
- `monte_carlo_worker.py` must preserve `Decimal` types until final aggregation
- Replace `float()` casts with proper Decimal arithmetic where precision matters

### R2: Decimal-Compatible Loss Generator Interface
- `ManufacturingLossGenerator.generate_losses()` should accept `Decimal` revenue
- Loss amounts should remain as `Decimal` throughout processing

### R3: Decimal-to-Float Conversion Point
- Define explicit conversion boundary (suggested: final result aggregation into numpy arrays)
- Document why conversion happens at this point
- Ensure conversion uses consistent precision (e.g., `float(quantize_currency(value))`)

### R4: Type Annotations
- Add proper type hints distinguishing `Decimal` vs `float` expectations
- Use `Union[Decimal, float]` only where both are truly acceptable

### R5: Backwards Compatibility
- Existing public APIs should continue to accept both `Decimal` and `float` inputs
- Use `to_decimal()` at API boundaries for normalization

## Implementation Steps

- [ ] **Phase 1: Monte Carlo Worker** (Critical Path)
  - [ ] 1.1 Refactor `monte_carlo_worker.py` to use `Decimal` for all financial calculations
  - [ ] 1.2 Keep `float()` conversion only for numpy array population at end
  - [ ] 1.3 Add `from ergodic_insurance.decimal_utils import to_decimal, ZERO` imports

- [ ] **Phase 2: Simulation Module**
  - [ ] 2.1 Review `simulation.py` for Decimal handling in `step_annual()`
  - [ ] 2.2 Ensure `SimulationResults` correctly handles Decimal-to-float conversion

- [ ] **Phase 3: Analysis Layer** (Lower Priority)
  - [ ] 3.1 Audit `business_optimizer.py` conversions - some may be intentional for numpy operations
  - [ ] 3.2 Audit `decision_engine.py` conversions
  - [ ] 3.3 Document which conversions are necessary vs. accidental

- [ ] **Phase 4: Testing**
  - [ ] 4.1 Add precision verification tests comparing Decimal vs float results
  - [ ] 4.2 Add property-based tests ensuring accounting identities hold
  - [ ] 4.3 Verify ledger balance verification passes with simulation results
  - [ ] 4.4 Fix all failing tests, even those unrelated to this issue

## Acceptance Criteria

- [ ] `monte_carlo_worker.py` performs all financial calculations in `Decimal` until final result
- [ ] `revenue_multiplier` calculation preserves precision: `to_decimal(revenue) / to_decimal(initial_revenue)`
- [ ] Insurance premium and loss calculations use `Decimal` arithmetic
- [ ] All `ledger.verify_balance()` calls pass after simulation runs
- [ ] No precision loss detected in 1000-year simulations (test: accounting equation holds exactly)
- [ ] Type hints accurately reflect actual types used at each point
- [ ] All tests pass, even those currently failing for unrelated reasons to this issue

## References

- Python `decimal` module: https://docs.python.org/3/library/decimal.html
- GAAP requirements for financial precision
- Related: `ergodic_insurance/decimal_utils.py` - existing utility functions
